.circleci/config.yml

```diff
@@ -217,7 +217,7 @@ jobs:
       - install_js
       - run:
           name: '`yarn prettier` changes committed?'
-          command: yarn prettier --check
+          command: yarn prettier --mode=check
```

Comment:
Shouldn't we opt for `check-staged` instead to mimic the current behavior and fast runtime? 🤔 
![Screenshot 2023-10-13 at 20 53 52](https://github.com/mui/material-ui/assets/4941090/0c17e55e-1fd9-46f9-a367-4c81e3e955e9)

Reply:
`check-changed` will miss potential problems when `prettier` updates. I'll run `check` when the lockfile has changed and `check-changed` in all other cases. What do you think?

Reply:
Good point! 💯 
Yeah, makes sense. If it's not too much trouble, run `check` when packages are updated, otherwise, stick with faster `check-changed`. 👌 

P.S. In regards to native JS or `zx` usage, I do not have a strong opinion either.
If given solution were taken from React repo, then I wouldn't discard them as sub-par options. 🤔